### PR TITLE
Add use_padded_fp flag to check pose/path services (#275)

### PR DESCRIPTION
**Pull request by corot**
Implementation of #275
@dorezyuk , please take a look and approve if the PR meets your feature request

### CODE REVIEW

File: mbf_costmap_nav/src/mbf_costmap_nav/costmap_navigation_server.cpp [hunk 1]
```diff
@@ -601,11 +601,12 @@ bool CostmapNavigationServer::callServiceCheckPoseCost(mbf_msgs::CheckPose::Requ
...
   double yaw = tf::getYaw(pose.pose.orientation);
 
-  // ensure costmap is active so cost reflects latest sensor readings
+  // ensure costmap is active so cost reflects the latest sensor readings
   costmap->checkActivate();
 
   // pad raw footprint to the requested safety distance; note that we discard footprint_padding parameter effect
-  std::vector<geometry_msgs::Point> footprint = costmap->getUnpaddedRobotFootprint();
+  std::vector<geometry_msgs::Point> footprint =
+      request.use_padded_fp ? costmap->getRobotFootprint() : costmap->getUnpaddedRobotFootprint();
```
Comment (dorezyuk) [inline mbf_costmap_nav/src/mbf_costmap_nav/costmap_navigation_server.cpp:609]: sounds good but could we add a getFootprint helper into the cpp with 
```cpp
static std::vector<geometry_msgs::Point> getFootprint(const costmap_2d::Costmap2DROS& costmap, const bool use_padded_fp){

}

```

**Comment by corot** [comment]
ping @dorezyuk....  come on,,, I'm implementing your feature request!!!
also @spuetz , can u take a look?

**Comment by dorezyuk** [comment]
@corot sorry sorry sorry, I reviewed it long time ago but missed the "submit review" button

File: mbf_costmap_nav/src/mbf_costmap_nav/costmap_navigation_server.cpp (hunk 1 above)
Comment (corot) [reply mbf_costmap_nav/src/mbf_costmap_nav/costmap_navigation_server.cpp:609]: > sounds good but could we add a getFootprint helper into the cpp with

eh....  no   :unamused:

come on, is gonna be a single line function!

Comment (dorezyuk) [reply mbf_costmap_nav/src/mbf_costmap_nav/costmap_navigation_server.cpp:609]: haha, ok if you don't like it that much

**Review by dorezyuk — APPROVED** [review]

**Review by spuetz — APPROVED** [review]

**Pull request merged**